cmd/gossamer/config.go

```diff
@@ -444,13 +467,49 @@ func setDotGlobalConfig(ctx *cli.Context, tomlCfg *ctoml.Config, cfg *dot.Global
 	}
 
 	cfg.NoTelemetry = ctx.Bool("no-telemetry")
+}
 
-	logger.Debug(
-		"global configuration",
-		"name", cfg.Name,
-		"id", cfg.ID,
-		"basepath", cfg.BasePath,
-	)
+func setDotGlobalConfigName(ctx *cli.Context, tomlCfg *ctoml.Config, cfg *dot.GlobalConfig) error {
+	globalBasePath := utils.ExpandDir(cfg.BasePath)
+	initialised := dot.NodeInitialized(globalBasePath, false)
+
+	thereIsNameTomlConfig := tomlCfg.Global.Name != ""
+	thereIsNameFlag := ctx.GlobalString(NameFlag.Name) != ""
+
+	// consider the --name flag as higher priority
+	if thereIsNameFlag {
+		cfg.Name = ctx.GlobalString(NameFlag.Name)
+		return nil
+	}
+
+	// consider the name on config as a second priority
+	if thereIsNameTomlConfig {
+		cfg.Name = tomlCfg.Global.Name
+		return nil
+	}
+
+	// if node was previusly initialised
```

Comment:
typo `previously`
Also, try to remove the new lines. Comments seem to be broken into multiple newlines.